services/placement/algo/sharded_helper.go

```diff
@@ -350,46 +392,50 @@ func (ph *placementHelper) returnInitializingShardsToSource(shardSet map[uint32]
 		if sourceID == "" {
 			continue
 		}
-		sourceInstance, ok := ph.instances[sourceID]
+		sourceInstance, ok := candidateMap[sourceID]
 		if !ok {
-			return fmt.Errorf("could not find sourceID %s in the placement", sourceID)
+			continue
```

Comment:
Basically it won't happen as the placement were always validated before reaching here, but I guess won't hurt to keep returning errors here.